### PR TITLE
Init commit

### PR DESCRIPTION
**Pull request by SanchayPahalwani**

### CODE REVIEW

File: CruxPay/Classes/CruxClient.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        let addressMap = try! JSONSerialization.jsonObject(with: data, options: .allowFragments) as! [String: Any]
+        self.jsBridge.executeAsync(method: "putAddressMap", params: [addressMap], onResponse: callback_fn(jsObj:), onErrorResponse: error_callback_fn(jsObj:))
+    }
+    
+    public func registerCruxID(cruxIDSubdomain: String, onResponse: @escaping () -> (), onErrorResponse: @escaping (CruxClientError) -> ()) {
+        func callback_fn (jsObj: JSValue) -> () {
+            if jsObj.isUndefined {
+                onResponse()
+            } else {
+                print(jsObj)
```
Comment (govind1246) [inline CruxPay/Classes/CruxClient.swift]: remove print

File: CruxPay/Classes/CruxJSBridge.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        
+        let consoleLog: @convention(block) (String) -> Void = { message in
+            print("console.log: " + message)
+        }
+        context?.setObject(unsafeBitCast(consoleLog, to: AnyObject.self),
+           forKeyedSubscript: "_consoleLog" as NSCopying & NSObjectProtocol)
+        _ = context?.evaluateScript("var window = this;")
+        _ = context?.evaluateScript(requestDeps)
+        _ = context?.evaluateScript(promiseDeps)
+        JSFetch.provideToContext(context: context!, hostURL: "https://www.cruxpay.com")
```
Comment (govind1246) [inline CruxPay/Classes/CruxJSBridge.swift]: why do we need cruxpay,.com inside sdk?

File: CruxPay/Classes/CruxJSBridge.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    
+    init(configBuilder: CruxClientInitConfig.Builder) {
+        prepareCruxClientInitConfig(configBuilder: configBuilder)
+        context?.evaluateScript("cruxClient = new window.CruxPay.CruxClient(cruxClientInitConfig)")
+        context?.evaluateScript("""
+            cruxClient.init()
+            .then(() => {
+                console.log('CruxClient initialized');
+            }).catch((err) => {
+                console.log('CruxClient error', err);
```
Comment (govind1246) [inline CruxPay/Classes/CruxJSBridge.swift]: error is being suppressed here.

File: CruxPay/Classes/CruxJSBridge.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                                 forKeyedSubscript: "_getRandomValues" as (NSCopying & NSObjectProtocol)?)
+        _ = context?.evaluateScript(cruxJS)
+        
+        return context
+    }()
+    
+    init(configBuilder: CruxClientInitConfig.Builder) {
+        prepareCruxClientInitConfig(configBuilder: configBuilder)
+        context?.evaluateScript("cruxClient = new window.CruxPay.CruxClient(cruxClientInitConfig)")
+        context?.evaluateScript("""
```
Comment (govind1246) [inline CruxPay/Classes/CruxJSBridge.swift]: Is this an async execution or sync execution? In case its async then how does caller know about completion?

File: CruxPay/Classes/CruxJSBridge.swift [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        """)
+    }
+    private func prepareCruxClientInitConfig(configBuilder: CruxClientInitConfig.Builder) -> Void {
+        let cruxClientInitConfig: CruxClientInitConfig  = configBuilder.create();
+        var cruxClientInitConfigString: String
+        cruxClientInitConfigString = cruxClientInitConfig.getCruxClientInitConfigString()!;
+        if (!cruxClientInitConfigString.isEmpty) {
+            context?.evaluateScript("cruxClientInitConfig = \(cruxClientInitConfigString);")
+            context?.evaluateScript("cruxClientInitConfig['storage'] = inmemStorage;")
+            context?.evaluateScript("cruxClientInitConfig['getEncryptionKey'] = function() { return 'fookey';}")
```
Comment (govind1246) [inline CruxPay/Classes/CruxJSBridge.swift]: replace fookey with some random and long string.

File: Example/CruxPay.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -0,0 +1,389 @@
+// !$*UTF8*$!
+{
```
Comment (govind1246) [inline Example/CruxPay.xcodeproj/project.pbxproj:2]: is this file required?

File: Example/CruxPay.xcodeproj/xcshareddata/xcschemes/CruxPay-Example.xcscheme [hunk 7]
```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Scheme
+   LastUpgradeVersion = "0900"
+   version = "1.3">
```
Comment (govind1246) [inline Example/CruxPay.xcodeproj/xcshareddata/xcschemes/CruxPay-Example.xcscheme:4]: is this required?

File: Example/Pods/CryptoSwift/Sources/CryptoSwift/BlockMode/CBC.swift [hunk 8]
```diff
@@ -0,0 +1,70 @@
+//
+//  CryptoSwift
```
Comment (govind1246) [inline Example/Pods/CryptoSwift/Sources/CryptoSwift/BlockMode/CBC.swift:2]: whats the use of these CBC, CTR, CFB, CCM etc?

File: CruxPay/Classes/CruxJSBridge.swift [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        
+        let frameworkBundle = Bundle(for: type(of: self))
+        let bundleURL = frameworkBundle.resourceURL!.appendingPathComponent("CruxPay.bundle")
+        let resourceBundle = Bundle(url: bundleURL)
+        
+        guard let cruxJSPath = resourceBundle?.path(forResource: "cruxpay-0.1.5", ofType: "js"),
+            let requestDepsPath = resourceBundle?.path(forResource: "requestDeps", ofType: "js"),
+            let promiseDepsPath = resourceBundle?.path(forResource: "promiseDeps", ofType: "js") else {
+                print("unable to read resource files.")
+                return nil
```
Comment (quixote911) [inline CruxPay/Classes/CruxJSBridge.swift]: Raise error here if we can't read resource files.

File: CruxPay/Classes/DataTypes.swift [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        }
+        public func create() -> CruxClientInitConfig {
+            return (CruxClientInitConfig(builder: self))
+        }
+    }
+    
+    private var builder: Builder;
+    private var privateKey: String?;
+    private var walletClientName: String;
+    private var _encryptionKeyValue: String;
```
Comment (quixote911) [inline CruxPay/Classes/DataTypes.swift]: This is not being used, remove

**Review by quixote911 — CHANGES_REQUESTED** [review]

File: CruxPay/Classes/CruxJSBridge.swift (hunk 9 above)
Comment (SanchayPahalwani) [reply CruxPay/Classes/CruxJSBridge.swift]: Call can throw, but errors cannot be thrown out of a property initializer

Initializing context here

File: CruxPay/Classes/DataTypes.swift (hunk 10 above)
Comment (SanchayPahalwani) [reply CruxPay/Classes/DataTypes.swift]: Removed

**Comment by quixote911** [comment]
Please add some basic instructions to the Readme - on how to use and run the package.

1. SDK Initialisation
2. getCruxIDState
3. resovleAddressForCruxID

**Review by quixote911 — APPROVED** [review]
LGTM
